component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -164,6 +169,30 @@ public boolean isValidAssertion(SignedJWT signedJWT) throws OAuthClientAuthnExce
                 issuedTime = issuedAtTime.getTime();
             }
 
+            //   Obtain the signing algorithm used to sign the JWT in the request.
+            String requestSigningAlgorithm = signedJWT.getHeader().getAlgorithm().getName();
+            if (!isValidSignatureAlgorithm(requestSigningAlgorithm, consumerKey)) {
+                throw new OAuthClientAuthnException("Signature algorithm used in the request is invalid.",
+                        OAuth2ErrorCodes.INVALID_CLIENT);
+            }
+
+            /* For FAPI compliant applications the allowed JWT signing algorithm should be configured at the
```

Comment:
Looks like this comment needs to be updated with the code changes that have been done.

Reply:
Fixed in e9d0e7eeb2260413659c89164eed65584a7d4278